#
# makes lsh compile with gmp3
#
# make[7]: Entering directory
# `/home/frogger/projects/ptxdist/ptxdist-i586-glibc/build/lsh-1.5.3/src/nettle/testsuite'
# i386-linux-gcc -ggdb3 -Wall -W -Wmissing-prototypes
# -Wmissing-declarations -Wstrict-prototypes -Wpointer-arith
# -Wbad-function-cast -Wnested-externs -L..  -o rsa-encrypt-test
# rsa-encrypt-test.o testutils.o -lnettle ../libnettle.a -lgmp
# rsa-encrypt-test.o: In function `test_main':
# /home/frogger/projects/ptxdist/ptxdist-i586-glibc/build/lsh-1.5.3
# /src/nettle/testsuite/rsa-encrypt-test.c:41:
# undefined reference to `gmp_fprintf'
#
# http://www.mail-archive.com/lsh-bugs@lists.lysator.liu.se/msg00121.html
#
diff -ruN lsh-1.5.3-orig/src/nettle/testsuite/rsa-encrypt-test.c lsh-1.5.3/src/nettle/testsuite/rsa-encrypt-test.c
--- lsh-1.5.3-orig/src/nettle/testsuite/rsa-encrypt-test.c	Tue Aug 26 08:44:51 2003
+++ lsh-1.5.3/src/nettle/testsuite/rsa-encrypt-test.c	Sat Oct 25 06:01:09 2003
@@ -38,7 +38,11 @@
 		     gibberish));
 
   if (verbose)
-    gmp_fprintf(stderr, "encrypted: %Zd\n", gibberish);
+    {
+      /* In which GMP version was gmp_fprintf introduced? */
+      fprintf(stderr, "encrypted: ");
+      mpz_out_str(stderr, 10, gibberish);
+    }
   
   decrypted = alloca(msg_length + 1);
 
